materializer/nestedfieldselection/README.md

```diff
@@ -0,0 +1,132 @@
+`@materializer` is a great way to seamlessly extend a type with data from another type by executing an operation, sometimes with arguments supplied by the extended type. 
+
+Sometimes the data needed is nested, so @materializer provides the ability to 
+specify a single field selection path.
+
+Assume that the original schema contains documents of type `document` with `content` and `author`, and that `content` has a required identifier provided by any request, and that the query operation only exposes direct access to the document, and not to the individaul `content` and `author`:
+
+```
+type Document {
+   docId: String
+   content(id: String!): Content
+     @materializer(query: "content", arguments:[{name: "id" argument:"id"}])
+   author: Author 
+     @materializer(query: "author")
+   lastUpdate: Date
+}
+
+type Content @mock {
+  id: String!
+  text: String!
+  summary: String!
+}
+
+type Author {
+    name: String!
+    country: String
+}
+
+type Query {
+  document:  Document
+       @rest(
+       endpoint: "stepzen:empty"
+       ecmascript: """
+        function transformREST() { 
+            const data = {
+                author: {
+                            name: 'Roy Derks',
+                            country: 'Netherlands'
+                        }
+                        };
+                const resp = JSON.stringify(data);
+                return resp
+            }
+     """
+       )
+}
+```
+
+The schema evolves to represent books with names and a reference to the document content from the previous types. Perhaps these types come from different backends. 
+
+```
+type Book {
+  refId: String!
+  name: String
+} 
+
+type Query {
+  book: Book
+    @rest(
+       endpoint: "stepzen:empty"
+       ecmascript: """
+        function transformREST() { 
+            const data = {
+                refId: '27',
+                name: 'Fullstack GraphQL'
+                };
+                const resp = JSON.stringify(data);
```

Comment:
minor point, but just `return JSON.stringify(data);` - cleaner.

Reply:
maybe the new jq/jsonata methods would be cleaner for return JSON data.

Reply:
going to give this a try in the api.graphql file @cckim .. Thanks!